File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        n=len(nums)
        l=0
        r=0
        z=0
        ans=0
        
        for r in range(n):
            if nums[r]==0:
                z+=1
            while z>1:
                if nums[l]==0:
                    z-=1
                l+=1
            ans=max(ans,r-l+1)
        return ans-1

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int l=0,r=0,ans=0,c0=0;
        for(;r<nums.length;r++){
            if(nums[r]==0){
                c0++;
            }
        for(;c0>1;l++){
            if(nums[l]==0){
                c0--;
            }}
            ans=Math.max(ans,r-l+1);
        }
        return ans-1;
    }
}